src/shims/foreign_items/posix.rs

```diff
@@ -313,11 +310,53 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_null(dest)?;
             }
 
-            // Incomplete shims that we "stub out" just to get pre-main initialization code to work.
+            // Threading
+            "pthread_create" => {
+                assert_eq!(args.len(), 4);
+                let result = this.pthread_create(args[0], args[1], args[2], args[3])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_join" => {
+                assert_eq!(args.len(), 2);
+                let result = this.pthread_join(args[0], args[1])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_detach" => {
+                assert_eq!(args.len(), 1);
+                let result = this.pthread_detach(args[0])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+            "pthread_self" => {
+                assert_eq!(args.len(), 0);
+                this.pthread_self(dest)?;
+            }
+            "prctl" => {
+                assert_eq!(args.len(), 5);
+                let result = this.prctl(args[0], args[1], args[2], args[3], args[4])?;
+                this.write_scalar(Scalar::from_i32(result), dest)?;
+            }
+
+            "pthread_attr_getguardsize" => {
+                assert_eq!(args.len(), 2);
+
+                let guard_size = this.deref_operand(args[1])?;
+                let guard_size_type = args[1].layout.ty
+                    .builtin_deref(true)
+                    .ok_or_else(|| err_ub_format!(
+                        "wrong signature used for `pthread_attr_getguardsize`: first argument must be a raw pointer."
+                    ))?
+                    .ty;
+                let guard_size_layout = this.layout_of(guard_size_type)?;
+                this.write_scalar(Scalar::from_uint(crate::PAGE_SIZE, guard_size_layout.size), guard_size.into())?;
```

Comment:
Why do you have to determine the size in this complicated way from the type? Is it not always the same, or always the pointer size?